lib/runCall.js

```diff
@@ -53,240 +55,93 @@ module.exports = function (opts, cb) {
+    blockHeader.setNumber(block.header.number)
+    blockHeader.setGaslimit(block.header.gasLimit)
+    blockHeader.setUnixtimestamp(timestamp)
+    return blockHeader
   }
 
-  function runCode (cb) {
-    if (!code) {
-      vmResults.exception = 1
-      stateManager.commit(cb)
-      return
-    }
-
-    var runCodeOpts = {
-      code: code,
-      data: txData,
-      gasLimit: gasLimit,
-      gasPrice: gasPrice,
-      address: toAddress,
-      origin: origin,
-      caller: caller,
-      value: txValue.toArrayLike(Buffer),
-      block: block,
-      depth: depth,
-      selfdestruct: selfdestruct,
-      static: isStatic,
-      storageReader: storageReader
-    }
-
-    // run Code through vm
-    var codeRunner = isCompiled ? self.runJIT : self.runCode
-    codeRunner.call(self, runCodeOpts, parseRunResult)
-
-    function parseRunResult (err, results) {
-      vmResults = results
-
-      if (createdAddress) {
-        // fee for size of the return value
-        var totalGas = results.gasUsed
-        if (!results.runState.vmError) {
-          var returnFee = new BN(results.return.length * self._common.param('gasPrices', 'createData'))
-          totalGas = totalGas.add(returnFee)
-        }
-        // if not enough gas
-        if (totalGas.lte(gasLimit) && (self.allowUnlimitedContractSize || results.return.length <= 24576)) {
-          results.gasUsed = totalGas
-        } else {
-          results.return = Buffer.alloc(0)
-          // since Homestead
-          results.exception = 0
-          err = results.exceptionError = ERROR.OUT_OF_GAS
-          results.gasUsed = gasLimit
-        }
-      }
-
-      gasUsed = results.gasUsed
-      if (err) {
-        results.logs = []
-        stateManager.revert(function (revertErr) {
-          if (revertErr || !isCompiled) cb(revertErr)
-          else {
-            // Empty precompiled contracts need to be deleted even in case of OOG
-            // because the bug in both Geth and Parity led to deleting RIPEMD precompiled in this case
-            // see https://github.com/ethereum/go-ethereum/pull/3341/files#diff-2433aa143ee4772026454b8abd76b9dd
-            // We mark the account as touched here, so that is can be removed among other touched empty accounts (after tx finalization)
-            if (err === ERROR.OUT_OF_GAS || err.error === ERROR.OUT_OF_GAS) {
-              stateManager.getAccount(toAddress, (getErr, acc) => {
-                if (getErr) cb(getErr)
-                else stateManager.putAccount(toAddress, acc, cb)
-              })
-            } else {
-              cb()
-            }
-          }
-        })
-      } else {
-        stateManager.commit(cb)
-      }
-    }
+  function createEthereumConfig () {
+    var ethereumConfig = new msg_pb.EthereumConfig()
+    ethereumConfig.setMaxcodesize(new Uint8Array([96, 0]))
+    ethereumConfig.setAccountstartnonce(new Uint8Array([0]))
+    ethereumConfig.setFrontierblocknumber(new Uint8Array([0]))
```

Comment:
how does ethereumjs-vm choose which hard fork to use?

Reply:
It looks like only three hardforks are currently supported: `byzantium`, `constantinople` and `petersburg`.  Ganache takes it as a command line argument at the startup https://github.com/ethereumjs/ethereumjs-vm/blob/b48b5c23524c41285f6e69667ca11b9e42133f21/lib/index.js#L41-L46

Reply:
In that case, let's leave this for now as it depends on updating the kevm plugin to support the new hardforks. We should put this on your list of tasks though.